src/bigdecimal.ts

```diff
@@ -663,13 +664,14 @@ export class BigDecimal {
         input: string,
         offset: number,
         len: number,
+        scale: number | undefined,
```

Comment:
> I think we can use setScale

I tried to use it but failed. I think passing the scale in `fromString` is still the right approach in this case (Mostly because `fromString` is an internal function).

The issue with using `setScale` after constructing the BigDecimal:
  - If the constructor scales up, the resulting value will simply add zeros behind the comma instead of shifting the integer to the right. 
  - If the constructor scales down, the constructor will throw as it requires rounding in some case

I added a few tests on `plainString` to illustrate it with my current solution.
With the `setScale` solution it would result to:
```ts 
const big = Big(Number.MAX_SAFE_INTEGER + 1, 1); 
// 9007199254740992.0 instead of 900719925474099.2

const big2 = Big(Number.MAX_SAFE_INTEGER + 1, -1); 
// setScale throws an error as it tries to transform `9007199254740992` to `9007199254740990` 
```

Reply:
I think you are right. I think your approach is fine. Should we document the new behaviour? I mean after this PR, BigDecimals built using unsafe numbers will be built as if they are passed as strings. I feel like we should mention this somewhere. Most probably the best place is the constructor documentation.

Reply:
addressed in 1147ff5baf4d245bfbe6d7a2b1e835b9b7b71b80